S2VX.Game/Story/S2VXStory.cs

```diff
@@ -87,6 +88,20 @@ public class S2VXStory : CompositeDrawable {
             Approaches.RemoveApproach(note);
         }
 
+        // Before starting the Story in the PlayScreen, we want to explicitly
+        // remove GameNotes up to some certain track time. This is so that we
+        // won't hear Miss hitsounds and prematurely calculate score.
+        public void RemoveNotesUpTo(double trackTime) {
+            while (Notes.Children.Count > 0 && Notes.Children.First().EndTime < trackTime) {
```

Comment:
I still see the issue of notes being counted as Misses in the case where you still have notes after trackTime. I breakpointed and investigated Notes.Children(). They are not ordered by EndTime, seems like they are arbitrarily ordered.